Backend/Backend/Infrastructure/FileStorageHelper.cs

```diff
@@ -0,0 +1,296 @@
+        
+        // Filename stored in the database
+        string userAvatarName = string.Format("{0}{1}{2}", avatarFileName, FILE_SPLIT_KEY, avatarFile.ContentType);
+
+        // Get the dir path
+        string dirPath = Combine(GetCurrentDirectory(), BASE_DIR, AVATARS_DIR_NAME);
+
+        // Make sure that the dir exists, otherwise create it
+        if(!Directory.Exists(dirPath))
+            CreateDirectory(dirPath);
+
+        // Get the file path
+        string filePath = Combine(dirPath, avatarFileName);
+
+        // Save the file
+        using FileStream fileStream = Create(filePath);
+        avatarFile.CopyTo(fileStream);
+
+        // Return the filename stored in the database
+        return userAvatarName;
+    }
+
+    /// <summary>
+    /// Removes a user avatar.
+    /// </summary>
+    /// <param name="userAvatarName"></param>
+    public static void RemoveUserAvatar(string userAvatarName)
+    {
+        // Get the file path
+        string userAvatarFilePath = GetUserAvatarPath(userAvatarName);
+
+        // Check if the file exists
+        if(File.Exists(userAvatarFilePath))
+        {
+            // Delete the file
+            File.Delete(userAvatarFilePath);
+        }
+    }
+
+    /// <summary>
+    /// Gets the path to a user avatar.
+    /// </summary>
+    /// <param name="userAvatarName"></param>
+    /// <returns></returns>
+    public static string GetUserAvatarPath(string userAvatarName)
+    {
+        return Combine(GetCurrentDirectory(), BASE_DIR, AVATARS_DIR_NAME, userAvatarName.Split(FILE_SPLIT_KEY)[0]);
+    }
+
+    #endregion
+
+    #region Podcast
+
+    /// <summary>
+    /// Saves a podcast cover art and returns the filename stored in the database.
+    /// </summary>
+    /// <param name="podcastId"></param>
+    /// <param name="coverArtFile"></param>
+    /// <returns></returns>
+    public static string SavePodcastCoverArt(Guid podcastId, IFormFile coverArtFile)
+    {
+        return SavePodcastCoverArt(podcastId.ToString(), coverArtFile);
+    }
+
+    /// <summary>
+    /// Saves a podcast cover art and returns the filename stored in the database.
+    /// </summary>
+    /// <param name="podcastId"></param>
+    /// <param name="coverArtFile"></param>
+    /// <returns></returns>
+    public static string SavePodcastCoverArt(string podcastId, IFormFile coverArtFile)
+    {
+        // Filename stored on the server filesystem
+        string coverArtFileName = string.Format("{0}.{1}", podcastId, coverArtFile.ContentType.Split('/')[1]);
+        
+        // Filename stored in the database
+        string coverArtName = string.Format("{0}{1}{2}", coverArtFileName, FILE_SPLIT_KEY, coverArtFile.ContentType);
+
+        // Get the dir path
+        string dirPath = Combine(GetCurrentDirectory(), BASE_DIR, PODCASTS_DIR_NAME);
+
+        // Make sure that the dir exists, otherwise create it
```

Comment:
is this necessary? we have another one right below

Reply:
oh that didnt comment correctly, I meant the create dir right below this

Reply:
You're right. Removed it.